engine/src/main/java/com/datatorrent/stram/plan/physical/StreamMapping.java

```diff
@@ -188,6 +192,69 @@ private void addSlidingUnifiers()
     }
   }
 
+  private List<PTOutput> setupPartitionerUnifiers(List<PTOutput> upstream, Partitioner<Operator> partitioner)
+  {
+    Operator.Unifier<?> unifier = streamMeta.getSource().getUnifier();
+
+    if (unifier == null) {
+      unifier = new DefaultUnifier();
+    }
+
+    Collection<Partitioner.Partition<Operator>> collection = new ArrayList<>(1);
+    DefaultPartition<Operator> firstPartition = new DefaultPartition<Operator>(unifier);
+    collection.add(firstPartition);
+
+    Collection<Partitioner.Partition<Operator>> partitions = partitioner.definePartitions(collection, new PhysicalPlan.UnifierPartitioningContextImpl());
+
+    if (partitions.size() <= 1) {
+      return upstream;
+    }
+    OperatorMeta om = streamMeta.getSource().getOperatorMeta();
+
+    InputPortMeta portMeta = null;
+    Map<InputPortMeta, StreamMeta> inputs = om.getInputStreams();
+    if (!inputs.isEmpty()) {
+      portMeta = inputs.keySet().iterator().next();
+      DefaultPartition.assignPartitionKeys(partitions, portMeta.getPortObject());
+    }
+
+    List<PTOutput> Unifierupstream = new LinkedList<>();
+    for (Partitioner.Partition<Operator> partition : partitions) {
+      PortMappingDescriptor mergeDesc = new PortMappingDescriptor();
```

Comment:
Can we move lines 223-254 into their own function? This will help make this function more readable.